itou/www/apply/forms.py

```diff
@@ -349,45 +349,65 @@ class SiaePrescriberFilterJobApplicationsForm(FilterJobApplicationsForm):
     Job applications filters common to SIAE and Prescribers.
     """
 
-    sender = forms.MultipleChoiceField(
-        required=False, label=_("Personne"), widget=Select2MultipleWidget
+    senders = forms.MultipleChoiceField(
+        required=False, label=_("Nom"), widget=Select2MultipleWidget
+    )
+
+    job_seekers = forms.MultipleChoiceField(
+        required=False, label=_("Candidat"), widget=Select2MultipleWidget
     )
 
     def __init__(self, job_applications_qs, *args, **kwargs):
         self.job_applications_qs = job_applications_qs
         super().__init__(*args, **kwargs)
-        self.fields["sender"].choices += self.get_sender_choices()
+        self.fields["senders"].choices += self._get_choices_for("sender")
+        self.fields["job_seekers"].choices = self._get_choices_for("job_seeker")
+
+    def _get_choices_for(self, user_type):
+        users = self.job_applications_qs.get_unique_fk_objects(user_type)
+        users = [user for user in users if user.get_full_name()]
+        users = [(user.id, user.get_full_name().title()) for user in users]
+        return sorted(users, key=lambda l: l[1])
+
+    def _humanize_multiple_choice_for_users(self, user_ids, field_name):
+        values = [
+            get_user_model().objects.get(id=int(user_id)).get_full_name().title()
```

Comment:
Depuis ce projet je me suis mis à utiliser `pk` plutôt que `id` et apparemment je suis pas le seul à suivre cette convention cf https://stackoverflow.com/questions/2165865/django-queries-id-vs-pk

donc `get(pk=int(user_id))`

Reply:
`pk` est un alias de la clé primaire. La clé primaire est nommée `id` par défaut mais elle peut avoir un autre nom :

- https://docs.djangoproject.com/en/dev/ref/models/instances/#the-pk-property
- https://docs.djangoproject.com/en/dev/topics/db/models/#automatic-primary-key-fields

J'aurais tendance à suivre la reco de @vgrange et d'utiliser `pk`

Reply:
Bien vu, merci ! Je change ça !